### DISCUSSION TITLE
Advanced Search for Issues Public Preview (Opt In roll out)

### DISCUSSION THREAD

**Discussion by github-staff** [community/community#139934] [Projects and Issues] [answered]
## Feedback wanted
Thank you for participating in the advanced search for issues public preview. Please leave your feedback below on what is working well, any bugs you encounter, and what else you’d like to see!

To provide your feedback on other experiences released at the same time, please visit:

- #139932
- #139933 
- #139935 
- #139936

## Writing advanced searches

From the repository Issues page, you can now build advanced queries using the `AND` and `OR` keywords, as well as nested queries using parentheses up to 5 layers deep. Here is the [existing search syntax documentation](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests) for reference.

![repo index page](https://github.com/user-attachments/assets/b21bc930-9bce-49f5-a2ac-fedd3117db6d)

This search allows you to find issues that are either bugs _or_ have high priority.

```
is:issue AND ( type:Bug OR label:priority-high )
```

This search allows you to find issues that were created in the last week that are _either_ bugs or tasks, _or_ medium priority ones you've authored.

```
is:issue AND created:>@today-1w AND (( type:Bug OR type:Task ) OR ( author:@me AND label:priority-medium ))
```
## Things to note
- Advanced search is not yet available in the API, but only through the UI from the repository issues page. We are working on providing this support.
- The `repo`, `org`, and `user` fields currently function as `OR` filters when a space is included between them, rather than an `AND` search as it works for other fields. These fields are not yet supported in nested queries.

## Looking for feedback

Please leave your feedback below 👇  on what is working well, what can be improved, and what else you’d like to see!

**Comment by htekdev** [comment]
Can we have this be usable in project queries as well...? Thats where its actually more useful vs in the repo issue query input:

![image](https://github.com/user-attachments/assets/82dfeef1-88a2-4e80-afbe-39005b361287)

**Reply by andrasjuhasz** [reply]
I thought the exact idea of the improvement is that these new search parameters can be used with Projects. I'm so disappointed. 🤦‍♂️

**Reply by Dyljyn** [reply]
Same here. I thought the update was for Projects. I did not even consider the search of a repository's issues list... 😅

**Reply by abaldwinls** [reply]
Also looking to have this updated in Projects, would be very helpful

**Reply by Jreevesjr** [reply]
Same!

**Reply by DanielNoord** [reply]
@ankneis Sorry for the direct ping, but you seem to consider this feature to be released according to your comment here: https://github.com/github/roadmap/issues/928

Now that I understand you were working on a very specific "Issues" project I understand that this might not have been part of the project scope and roadmap issue, but are there any plans to release this filtering for "Projects" as well? I never really need `AND` or `OR` in normal issue filtering, but I need it daily in my project views... Having inconsistent behaviour across these issue filter bars is quite confusing.

**Reply by abhaykhu** [reply]
Just noting that this feature, as is, is essentially useless for our org's workflows. It would however be very valuable in projects/project search.

**Reply by mamccorm** [reply]
+1 was excited to try this feature, but unfortunate to find out it's not yet supported for project queries. Is this on the future roadmap?

**Reply by jason-capsule42** [reply]
We are using these tools extensively on the Auro Design System team at Alaska Airlines. However, we rely primarly on the project views as each of our web components is in it's own repository. I was very jazzed about the advanced search features and equally deflated when I saw that they don't work in projects. I see this conversation goes back 3 months. Would love to know if project support is even being considered.

**Comment by rugk** [comment]
## UX: It's bad search shortcuts have been removed

I totally miss the "quick select" options that were there before. You could click on a dropdown on the left and choose "issues opened by you" (or similar). Literally, this is mostly the only option I have used the search bar there for. Like 80% of the time I likely want to (re-)find an open issue by my to check it's state, check for replies etc. (And yeah I cannot filter per repo in the notifications like that.) or e.g. to add new information I have found/debugged etc., when I don't know the title or so.
Long after that, "issues involving you" probably follow, for similar reasons. There no shortcut was there, but I would like it.

Old UI:
![image](https://github.com/user-attachments/assets/3fbe6682-9aa2-4e27-b09e-ba48354adb5b)


It was very convenient and easy (two clicks -> dropdown & clicking on the button). For the same process I now need to:
1. Click into the search bar.
2. Enter "a" at least. (Also note you need to know this.)
3. Click on "author:" (or write it I guess)
4. Click on "Me":
    ![image](https://github.com/user-attachments/assets/dcdf2380-7982-4426-89ba-14030a3adff5)

These are 3 clicks and one keypress at minimum to achieve the same repeating thing that previously was just two clicks away. I don't like this at all.
After all advanced search fine, but I mostly don't use or need any "advanced" features, I mostly just use plain old simple filtering and these should be as simple as before.

**Reply by rugk** [reply]
Also linked/cross-posted in https://github.com/orgs/community/discussions/139935#discussioncomment-10851387

**Reply by rileybroughten** [reply]
Thank you for sharing this feedback, as noted on the other discussion we're looking into it 👍

**Comment by gustavovalverde** [comment]
Allow to save an advanced query for future use. I'd like to have some favorite queries that I might be using on a daily or weekly basis. Having an option to re-use those would be great, so we don't have to rebuild them every time, or depend on our browser to bookmark them :)

**Reply by rileybroughten** [reply]
Noted this feedback, thank you!

**Reply by mavericohm** [reply]
A bit more feedback here on why bookmarking advanced searches is critical. One key use case is triaging issues that don’t yet have a project assigned. Advanced search enables us to add criteria not available in project views, like filtering for all issues without a project. This is essential for handling incoming bugs or alerts created by tools like Datadog or Freshdesk. These issues are added directly to the repository but don’t automatically get assigned to a project.

Without a way to bookmark this search view, we lose efficiency in managing these unassigned issues. Bookmarking would allow us to treat this view as a triage dashboard, making it easy to regularly revisit, organize, and assign these issues to the right projects. It’s a simple feature that could have a massive impact on improving workflows for teams managing high volumes of incoming requests.

**Comment by mayank99** [comment]
It took me some fiddling to figure out how to exclude "sub-issues".

Using the new filter UI (type into the search field to trigger the dropdpwn, then click "Exclude", then click "Sub-issue") generates the query `-sub-issue:` which doesn't seem to work. I also tried `-sub-issue` (without the colon at the end) with no luck.

![](https://github.com/user-attachments/assets/4b279e32-c657-4b62-93f1-e3d14d67724c)

What ultimately worked is `no:parent-issue`, but "No" is not surfaced in the initial filter UI, and it also seems redundant with "Exclude".

![](https://github.com/user-attachments/assets/dc5e2301-7fb8-4859-8930-9f7706de8815)

Side note: I think sub-issues should be excluded by the default filter.

**Reply by rileybroughten** [reply]
Thank you for this feedback, we have it noted! Correct that once you type `n` you will see the `No` dropdown appear. 
<img width="432" alt="Screenshot 2024-10-10 at 10 01 28 AM" src="https://github.com/user-attachments/assets/87c0304f-430d-4c17-8326-dfdb69b64501">

**Comment by steven-pribilinskiy** [comment]
Please collaborate with Copilot team to support natural language queries

![chrome_6I82U6k1ri](https://github.com/user-attachments/assets/0aa79ab1-a05c-45ad-9407-e88ff8d93bcf)

**Comment by matthiaskrgr** [comment]
Searching for multi-word strings no longer works properly.

When I am logged off / in an alt browser and do something like this
https://github.com/rust-lang/rust/issues?q=is%3Aissue%20state%3Aopen%20%20%22unexpected%20const%22%20
`is:issue state:open  "unexpected const" `
I get 16 results (I want to find issues that match the entire string somewhere)

When I am logged in (I assume this enable the new advanced search)  and open the same link, I get 270 results, most are which are useless, as if the query was `is:issue AND state:open AND ("unexpected" OR "const")` or something like that?

**Reply by rileybroughten** [reply]
Thank you for reporting this! We'll look into this and will provide an update when we have more details 👍

**Reply by rjgotten** [reply]
> Searching for multi-word strings no longer works properly.

Heck-- searching for **single-word** strings no longer works properly.
It returns _completely_ unrelated results where the word in question occurs _nowhere_ in the issue.
Search is basically broken.

[edit]
Actually-- it appears it isn't broken. It's just searching deep into collapsed `<details>` elements as are commonly used for listing technical details in bug reports, like what software users have installed on their environment; technical logs; etc.
Maybe **not** search and index in those?
Or at least not return results based on those _by default_ ?

**Reply by L-Clouds** [reply]
Looks like quoted phrases are finally working!

**Reply by rileybroughten** [reply]
We just pushed a fix this week! Let us know if you encounter this in the future 👍

**Comment by rbanks54** [comment]
Is advanced search going to be available in PRs tab as well?

Obviously I can use `is:pr` from the issues tab, but that's a bit annoying to do and isn't the best UX.

**Reply by david-crespo** [reply]
`is:pr` is a nice tip, thanks!

**Comment by rbanks54** [comment]
The `team-review-requested` autocomplete doesn't show all teams and any filtering seems to only be applied to the initially retrieved list, not everything that is available.

For example, I have a set of nested teams as follows
![image](https://github.com/user-attachments/assets/4a1b87c6-a3a8-44a9-91cd-9f4a7c5c8b08)

When I search for "teamsix" it doesn't appear in the list:
<img width="400" alt="image" src="https://github.com/user-attachments/assets/89523c9b-e9cb-416a-8171-6282df97c33a">

If I type the team name in the search returns results as expected.

I've also noticed that if I search for `assignee:rich` that _sometimes_ I see my name listed, but often I don't. Unfortunately, I haven't been able to consistently replicate this behaviour :(

**Comment by winniehell** [comment]
Being able to filter by issues that I have given a :+1: would be nice.

**Reply by rileybroughten** [reply]
Thank you for sharing this feedback! Noting that you can [filter by the number of reactions](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests#search-by-number-of-reactions), just not the specific reaction type 😄 
<img width="477" alt="Screenshot 2024-10-22 at 2 13 55 PM" src="https://github.com/user-attachments/assets/9fa84ea3-f43c-463f-9828-8b24ba86471d">

**Reply by winniehell** [reply]
When I want to get a list of issues that I'm interested in, my own reaction is far more helpful than the number of reactions from other people.
This is especially useful for open source projects with a slow release cycle where I don't want to join the discussion (so subscribing is not an option) but check regularly whether certain bugs have been fixed.

**Comment by L-Clouds** [comment]
When I try to perform a search that excludes certain text (plain words or quoted phrases), it seems to always show no results.  I've tried `NOT wordtoexclude`, `NOT "wordtoexclude"`, `-wordtoexclude`, `-"wordtoexclude"`, and (having read [this other comment](https://github.com/orgs/community/discussions/139934#discussioncomment-10896493)) `no:wordtoexclude`, `no:"wordtoexclude"` these latter two actually providing warnings of being invalid values for `no:`

![image](https://github.com/user-attachments/assets/7e085a51-2ebb-4555-8d9e-7eb5c1ae93f9)

**Reply by rileybroughten** [reply]
We are looking into a bug with the `NOT` keyword and how it does not return expected results. I will provide an update when we have a fix for this!

**Reply by L-Clouds** [reply]
Any update on this?

**Reply by rileybroughten** [reply]
Not quite yet, but I'll be sure to post here when this is resolved!

**Reply by rileybroughten** [reply]
We just pushed a fix on this, @L-Clouds could you try again and let us know if you are still seeing this issue?

**Reply by L-Clouds** [reply]
Oh, I didn't see the reply just above when I came to write this:

Discovered that there's been progress on this!  It looks like `NOT` works again, now, although it doesn't work for quoted phrases, instead using only the first word therein - but that looks like a separate issue covered under this comment: https://github.com/orgs/community/discussions/139934#discussioncomment-10930535

**Reply by rileybroughten** [reply]
Yes we're working on a fix for that one currently, expect to see this fixed soon 😁 I'll post an update when it's resolved!

**Comment by phyominoo993** [comment]
> ## Feedback wanted
> Thank you for participating in the advanced search for issues public preview. Please leave your feedback below on what is working well, any bugs you encounter, and what else you’d like to see!
> 
> To provide your feedback on other experiences released at the same time, please visit:
> 
> * [Sub-issues Public Preview #139932](https://github.com/orgs/community/discussions/139932)
> * [Issue Types Public Preview #139933](https://github.com/orgs/community/discussions/139933)
> * [Issues: Index, Create and Show Page Updates Public Preview #139935](https://github.com/orgs/community/discussions/139935)
> * [Increased Project Item Limits Public Preview #139936](https://github.com/orgs/community/discussions/139936)
> 
> ## Writing advanced searches
> From the repository Issues page, you can now build advanced queries using the `AND` and `OR` keywords, as well as nested queries using parentheses up to 5 layers deep. Here is the [existing search syntax documentation](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests) for reference.
> 
> ![repo index page](https://private-user-images.githubusercontent.com/101840513/364129174-b21bc930-9bce-49f5-a2ac-fedd3117db6d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA0NzAyMTIsIm5iZiI6MTczMDQ2OTkxMiwicGF0aCI6Ii8xMDE4NDA1MTMvMzY0MTI5MTc0LWIyMWJjOTMwLTliY2UtNDlmNS1hMmFjLWZlZGQzMTE3ZGI2ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwMVQxNDA1MTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNjE3ZTRlMDBjM2UxY2I5ZjdhZjg0OTBiMmUzMzlkMGEwNGZjYjBiMTgwNzNhOGU2OTQ3OTFiZWI2NDM4MTBlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.lAzWCkQW4BcJYMpYGzeznnnZjZ2YCE9Xwirb_vWXop0)
> 
> This search allows you to find issues that are either bugs _or_ have high priority.
> 
> ```
> is:issue AND ( type:Bug OR label:priority-high )
> ```
> 
> This search allows you to find issues that were created in the last week that are _either_ bugs or tasks, _or_ medium priority ones you've authored.
> 
> ```
> is:issue AND created:>@today-1w AND (( type:Bug OR type:Task ) OR ( author:@me AND label:priority-medium ))
> ```
> 
> ## Things to note
> * Advanced search is not yet available in the API, but only through the UI from the repository issues page. We are working on providing this support.
> * The `repo`, `org`, and `user` fields currently function as `OR` filters when a space is included between them, rather than an `AND` search as it works for other fields. These fields are not yet supported in nested queries.
> 
> ## Looking for feedback
> Please leave your feedback below 👇 on what is working well, what can be improved, and what else you’d like to see!

**Comment by thibaudcolas** [comment]
I think it’s missing a "help" link to view the advanced search syntax, like there was in the past. For example I only learned it supports reactions count filtering by looking at this thread :)

Otherwise this seems to be working great!

**Comment by rjgotten** [comment]
Searching for basic single-word queries is broken.
E.g. filtering for prior reported issues related to "SCSS" in VSCode:

![image](https://github.com/user-attachments/assets/961587a3-c517-4737-9905-09a501b60df7)
![image](https://github.com/user-attachments/assets/61f31c35-9ff1-4b36-aa71-be66d97e18d6)

Actual occurence of the search term is _zero_ in the issue title or body itself - and even the **entire** thread of discussion on it.

**Reply by rjgotten** [reply]
[edit] never mind.
It appears that the search term is actually present in the collapsed extensions in the opening post.
Looks like this is a thing where Firefox does not offer search through collapsed text.

With that, maybe another suggestion is in order:
Don't search in collapsed `<details>` elements by default - because most of the time you're going to get this kind of thing where tangentially related stuff is going to be present in a log file and it's just going to lead to a lot of false positive hits.

**Comment by htekdev** [comment]
Im finding that the query does not work in general search.

So here is an overview...
 
1. Search in Repository Issue Page - UI Highlighting✔️, Search Results ✔️ 
2. Search in Project - UI Highlighting ❌, Search Results ❌ 
3. Search in Search - UI Highlighting✔️, Search Results ❌

**Comment by htekdev** [comment]
In projects I am able to perform a filter `has:type` but in repo search and general search that is not available... I am forced to do `type:Task OR type:Bug OR ...` ect... I also tried `-no:type` but that didnt work properly either.

There is a general theme of inconsistency in functionality between Repo Issue Page, Projects, and Search from my [other comment](https://github.com/orgs/community/discussions/139934#discussioncomment-11246358)

**Reply by rileybroughten** [reply]
We now support `has:type` as a filter from the repository page. We're working on updating the documentation to reflect this. Let us know if you see any issues.

<img width="443" alt="Screenshot 2024-11-14 at 5 38 57 PM" src="https://github.com/user-attachments/assets/4baa89a4-fbfd-4615-a1a7-28ff6f3301bf">

**Comment by htekdev** [comment]
Im getting the following when trying to apply search for issues at org level...

![image](https://github.com/user-attachments/assets/6c4215ad-7c7c-42bb-bf1c-e8ad963450f5)

Im trying to apply this search via API and not getting any success. I see the note in the body of the discussion but wanting to know when this is going to be supported since without it will hinder my ability to take advantage of this awesome feature.

**Reply by rileybroughten** [reply]
I'll provide an update when we support advanced issue searching via the API 👍 Thank you for your patience on this!

**Comment by JorenVsyntra** [comment]
Hey everyone!

**Comment by jmssltd** [comment]
Having just switched to using Iterations in Projects for Sprints vs Milestone (which is repo based) for Sprints - being able to search across the Organization for Repo and Projects based fields will be a huge step for our business. Currently it appears that searching for Issues assigned to a Project Iteration (Sprint) does not work - using e.g. org:{orgname} created:{daterange} sprint:{sprintname}

**Comment by github-staff** [comment, marked as answer]
Thank you all for taking the time to share your feedback in this thread. 🙇‍♀️ We’re excited to announce that we have [expanded the roll out ](https://github.blog/changelog/2025-01-13-evolving-github-issues-public-preview/)and made meaningful updates that we believe will enhance your experience. This is a direct reflection of the conversations we’ve had here, and we’re grateful for your role in this process.

As part of this rollout, we’ve created a **[new feedback discussion](https://github.com/orgs/community/discussions/148716)** to gather your thoughts on the latest updates and ensure we continue to improve. If you’ve already shared your experiences here, we’ve taken your feedback on board. However, if you feel there are points worth revisiting or emphasizing, we encourage you to highlight them again in the new discussions to keep the conversation fresh and relevant.

We look forward to hearing more of our users exploring the evolution of GitHub issues and sharing their thoughts in the [new threads](https://github.com/orgs/community/discussions/148713)! See you there. 👋
